projects/core/src/error-handling/utils/error-interceptor-utils.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChainedErrorInterceptorFn,
+  ErrorInterceptor,
+  ErrorInterceptorPriority,
+} from '../error-interceptors/error-interceptor';
+
+/**
+ * Sorts error interceptors based on priority.
+ *
+ * @param errorInterceptors - error interceptors to sort
+ * @returns sorted error interceptors
+ *
+ * @internal
+ */
+export const sortErrorInterceptors = (
+  errorInterceptors: ErrorInterceptor[]
+): ErrorInterceptor[] => {
+  const highPriorityErrorInterceptors = new Array<ErrorInterceptor>();
+  const lowPriorityErrorInterceptors = new Array<ErrorInterceptor>();
+  const noPriorityErrorInterceptors = new Array<ErrorInterceptor>();
+
+  errorInterceptors.forEach((handler) => {
+    if (handler.priority === ErrorInterceptorPriority.HIGH) {
+      highPriorityErrorInterceptors.unshift(handler);
+    } else if (handler.priority === ErrorInterceptorPriority.LOW) {
+      lowPriorityErrorInterceptors.push(handler);
+    } else {
+      noPriorityErrorInterceptors.push(handler);
+    }
+  });
+  return [
+    ...highPriorityErrorInterceptors,
+    ...noPriorityErrorInterceptors,
+    ...lowPriorityErrorInterceptors,
+  ];
+};
+
+/**
+ * Handles error interceptors chain.
+ * @param next - next chained interceptor function that handles the error and calls the next interceptor
+ * @param interceptor - current interceptor
+ * @returns chained interceptor function
+ *
+ * @internal
+ */
+export const handleInterceptors = (
+  next: ChainedErrorInterceptorFn,
+  interceptor: ErrorInterceptor
+): ChainedErrorInterceptorFn => {
+  return (error: Error) => {
+    interceptor.interceptError(error, next);
+  };
+};
+
+//TODO: discuss whether we need to do anything specific at the end of the chain
+// maybe this is a good place to forward an error to the ExpressJS?
+/**
+ * The last function in the chain. Does nothing.
+ * @param _error - error
+ *
+ * @internal
+ */
+export const tailChain: ChainedErrorInterceptorFn = (_error: Error) => {
+  // do nothing
+};
```

Comment:
[MEDIUM]
If we have a flexible mechanism for adding new interceptors at any place, then I believe we don't need a hardcoded `tailChain` const.

Reply:
Ok, so let's keep this function empty.